### PR TITLE
Migrate old ut

### PR DESCRIPTION
**Pull request by shargon**
Last migration unit test part, it will be divided in different pr because it's too long, included in this PR:
- Migrate BasicBlock optimizer unit tests.
- Moved old test to OldEngine namespace.
- Created artifacts for future PR.
- Separate contracts that are designed not to be compiled into folders.

### CODE REVIEW

File: tests/Neo.Compiler.CSharp.UnitTests/TestCleanup.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+            // Define paths
+
+            string testContractsPath = Path.GetFullPath("../../../../Neo.Compiler.CSharp.TestContracts/Neo.Compiler.CSharp.TestContracts.csproj");
+            string artifactsPath = Path.GetFullPath("../../../TestingArtifacts");
+            var root = Path.GetPathRoot(testContractsPath) ?? "";
+
+            // Compile
+
+            var results = new CompilationEngine(new CompilationOptions()
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/TestCleanup.cs:40]: @Jim8y could you help me here? I'm not able to compile whole project together

> System.ArgumentException: SyntaxTree is not part of the compilation (Parameter 'syntaxTree')

Comment (Jim8y) [reply tests/Neo.Compiler.CSharp.UnitTests/TestCleanup.cs:40]: I will check.

Comment (shargon) [reply tests/Neo.Compiler.CSharp.UnitTests/TestCleanup.cs:41]: I think that is because there are contract designed to fail during compilation

**Comment by shargon** [comment]
@Jim8y could you help me here?

**Comment by shargon** [comment]
I will open a new one

**Pull request closed without merging**